p2p/transport/quic/conn_test.go

```diff
@@ -58,529 +66,571 @@ func runServer(t *testing.T, tr tpt.Transport, addr string) tpt.Listener {
 }
 
 func TestHandshake(t *testing.T) {
-	serverID, serverKey := createPeer(t)
-	clientID, clientKey := createPeer(t)
-	serverTransport, err := NewTransport(serverKey, nil, nil, nil)
-	require.NoError(t, err)
-	defer serverTransport.(io.Closer).Close()
-
-	handshake := func(t *testing.T, ln tpt.Listener) {
-		clientTransport, err := NewTransport(clientKey, nil, nil, nil)
-		require.NoError(t, err)
-		defer clientTransport.(io.Closer).Close()
-		conn, err := clientTransport.Dial(context.Background(), ln.Multiaddr(), serverID)
-		require.NoError(t, err)
-		defer conn.Close()
-		serverConn, err := ln.Accept()
-		require.NoError(t, err)
-		defer serverConn.Close()
-
-		require.Equal(t, conn.LocalPeer(), clientID)
-		require.True(t, conn.LocalPrivateKey().Equals(clientKey), "local private key doesn't match")
-		require.Equal(t, conn.RemotePeer(), serverID)
-		require.True(t, conn.RemotePublicKey().Equals(serverKey.GetPublic()), "remote public key doesn't match")
-
-		require.Equal(t, serverConn.LocalPeer(), serverID)
-		require.True(t, serverConn.LocalPrivateKey().Equals(serverKey), "local private key doesn't match")
-		require.Equal(t, serverConn.RemotePeer(), clientID)
-		require.True(t, serverConn.RemotePublicKey().Equals(clientKey.GetPublic()), "remote public key doesn't match")
-	}
+	for _, tc := range connTestCases {
```

Comment:
Please introduce a separate function, so you don't have to indent everything here.